tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -13,36 +13,74 @@
 * License for the specific language governing permissions and limitations
 * under the License.
 */
-#include "cmdlineparser.h"
 #include <boost/property_tree/json_parser.hpp>
 #include <boost/property_tree/ptree.hpp>
 #include <math.h>
 #include <sys/time.h>
+#include <getopt.h>
 #include <xcl2.hpp>
 
+const static struct option long_options[] = {{"path", required_argument, 0, 'p'},
+                                             {"device", required_argument, 0, 'd'},
+                                             {"iter_cnt", required_argument, 0, 'l'},
+                                             {"supported", no_argument, 0, 's'},
+                                             {0, 0, 0, 0}};
+
+static void printHelp() {
+    std::cout << "usage: %s <options>\n";
+    std::cout << "  -p <platform_test_area_path>\n";
+    std::cout << "  -d <device> \n";
+    std::cout << "  -l <loop_iter_cnt> \n";
+    std::cout << "  -s <check_supported>\n";
+}
+
 int main(int argc, char** argv) {
-    if (argc < 2) {
-        std::cout << "Usage: " << argv[0] << " <Platform Test Area Path>"
-                  << "<optional> -d device_id"
-                  << "<optional> -l iter_cnt" << std::endl;
+    int option_index = 0;
+    std::string dev_id = "0";
+    std::string test_path;
+    std::string iter_cnt = "10000";
+    std::string b_file = "/bandwidth.xclbin";
+    bool flag_s = false;
+    bool flag_p = false;
+    // Commandline
+    int c;
+    while ((c = getopt_long(argc, argv, "p:d:l:s", long_options, &option_index)) != -1) {
+        switch (c) {
+            case 'p':
+                test_path = optarg;
+                flag_p = true;
+                break;
+            case 'd':
+                dev_id = optarg;
+                break;
+            case 's':
+                flag_s = true;
+                break;
+            case 'l':
+                iter_cnt = atoi(optarg);
+                break;
+            default:
+                printHelp();
+                return 1;
+        }
+    }
+    if (!flag_p) {
```

Comment:
Instead of creating a flag, you can simply check if the str is empty or not
```suggestion
    if (test_path.empty()) {
```

Reply:
updated.